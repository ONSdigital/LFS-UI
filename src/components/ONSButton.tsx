import React, { Component } from 'react';

interface Props{
  label: string,
  id?: string,
  primary: boolean,
  small: boolean,
  onClick?: (...props: any[]) => void
}

export class ONSButton extends Component <Props, {}>{
  constructor(props : Props) {
    super(props);
  }

    render() {
    let className = "field btn " + (this.props.primary ? "" : "btn--secondary ") + (this.props.small ? "btn--small " : "");
    return (
      <button type="button" className={className} onClick={this.props.onClick}>
          <span className="btn__inner">{this.props.label}</span>
      </button>
    );
  }
}
