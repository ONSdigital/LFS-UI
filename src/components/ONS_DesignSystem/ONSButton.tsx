import React, {Component} from 'react';

interface Props {
    label: string,
    id?: string,
    primary: boolean,
    small: boolean,
    field?: boolean,
    loading?: boolean,
    marginRight?: number,
    onClick?: (...props: any[]) => void
}




export class ONSButton extends Component <Props, {}> {
    
    spacing = () => {
        let buttonStyle = {
            marginRight: String(this.props.marginRight) + "px",
          };
        return buttonStyle
    }

    render() {
        let className = "btn " + (this.props.loading ? "btn--loader is-loading " : "") + (this.props.field ? "field ": "") + (this.props.primary ? "" : "btn--secondary ") + (this.props.small ? "btn--small " : "");
        return (
            <button style={this.spacing()} type="button" disabled={this.props.loading} className={className} onClick={this.props.onClick} >
                <span className="btn__inner">{this.props.label}</span>
            </button>
        );
    }
}
