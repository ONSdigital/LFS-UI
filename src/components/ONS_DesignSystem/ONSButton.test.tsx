import React from 'react';
import { shallow, mount } from 'enzyme';
import {ONSButton} from "./ONSButton";
import sinon from 'sinon';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

describe("ONS Button Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    const buttonProps = {
        label: "Submit",
        primary: false,
        small: true,
        field: true,
        loading: false,
        onButtonClick: sinon.spy()
    }

    const loadingButtonProps = {
        label: "Submit",
        primary: true,
        small: false,
        onButtonClick: sinon.spy(),
        loading: true
    }

    function wrapper (render: any, buttonProps: any) {
        return render(
             <ONSButton label={buttonProps.label} 
                        id={buttonProps.id}
                        primary={buttonProps.primary}  
                        small={buttonProps.small} 
                        field={buttonProps.field}
                        loading = {buttonProps.loading}
                        marginRight = {buttonProps.marginRight}
                        onClick={buttonProps.onButtonClick}/>)
    }

    it("should render correctly", () => expect(wrapper(shallow, buttonProps).exists()).toEqual(true));

    it("should render with the correct label", () => {
        expect(wrapper(mount, buttonProps).find("ONSButton").getElement().props.label).toEqual(buttonProps.label);
    })
 
    it('simulates click events', () => {
        wrapper(mount, buttonProps).find('ONSButton').simulate('click');
        expect(buttonProps.onButtonClick).toHaveProperty('callCount', 1);
    })
 
    it('displays loading button', () => {
        expect(wrapper(mount, loadingButtonProps).find('button').hasClass('btn--loader is-loading ')).toEqual(true)
    })
     
})