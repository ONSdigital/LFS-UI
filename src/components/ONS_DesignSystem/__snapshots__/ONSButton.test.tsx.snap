// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Button Test matches Snapshot 1`] = `ShallowWrapper {}`;
